src/utils/__test__/SolutionsUtils.spec.js

```diff
@@ -95,3 +95,87 @@ describe('addRunTemplatesParametersIdsDict for a minimal or incomplete solution'
     });
   });
 });
+
+describe('patchIncompatibleValuesInSolution function unit tests', () => {
+  const TABLE_PARAMETER = {
+    varType: '%DATASETID%',
+    options: {
+      addRowEnabled: false,
+      subType: 'TABLE',
+      columns: [
+        {
+          field: 'name',
+          type: ['nonResizable', 'nonSortable'],
+          defaultValue: 'TKT',
+        },
+        {
+          field: 'age',
+          type: ['int'],
+          minValue: 0,
+          maxValue: 120,
+          acceptsEmptyFields: true,
+        },
+      ],
+    },
+  };
+  const PARAMETER_LIST = {
+    parameters: [TABLE_PARAMETER],
+  };
+  const PARAMETER_LIST_WITH_ADD_ROW = {
+    parameters: [{ ...TABLE_PARAMETER, options: { ...TABLE_PARAMETER.options, addRowEnabled: true } }],
+  };
+  const PARAMETER_LIST_WITH_NON_EDITABLE_COLUMN = {
+    parameters: [
+      {
+        ...TABLE_PARAMETER,
+        options: {
+          ...TABLE_PARAMETER.options,
+          columns: [
+            {
+              field: 'name',
+              type: ['nonResizable', 'nonSortable', 'nonEditable'],
+              defaultValue: 'TKT',
+            },
+            { ...TABLE_PARAMETER.options.columns[1] },
+          ],
+        },
+      },
+    ],
+  };
+  const PARAMETER_LIST_WITH_ALL_FEATURES = {
+    parameters: [
+      {
+        ...TABLE_PARAMETER,
+        options: {
+          ...TABLE_PARAMETER.options,
+          addRowEnabled: true,
+          columns: [
+            {
+              field: 'name',
+              type: ['nonResizable', 'nonSortable', 'nonEditable'],
+              defaultValue: 'TKT',
+            },
+            { ...TABLE_PARAMETER.options.columns[1] },
+            {},
+            null,
+            undefined,
+          ],
+        },
+      },
+    ],
+  };
+
+  test.each`
+    parameters                                 | expected                                   | warnCount
+    ${PARAMETER_LIST}                          | ${PARAMETER_LIST}                          | ${0}
+    ${PARAMETER_LIST_WITH_ADD_ROW}             | ${PARAMETER_LIST_WITH_ADD_ROW}             | ${0}
+    ${PARAMETER_LIST_WITH_NON_EDITABLE_COLUMN} | ${PARAMETER_LIST_WITH_NON_EDITABLE_COLUMN} | ${0}
+    ${PARAMETER_LIST_WITH_ALL_FEATURES}        | ${PARAMETER_LIST_WITH_ALL_FEATURES}        | ${1}
+  `('parse $parameters and fix it to get a good parameter list as $expected', ({ parameters, expected, warnCount }) => {
+    const warn = jest.spyOn(console, 'warn').mockImplementation(() => {});
+    SolutionsUtils.patchIncompatibleValuesInSolution(parameters);
+    expect(parameters).toStrictEqual(expected);
+    expect(warn).toHaveBeenCalledTimes(warnCount);
+    warn.mockReset();
```

Comment:
Something is odd in this test, you're always providing the same value in `parameters` and `expected` and then comparing that both values are "strict equal", which will always be true :slightly_smiling_face: 

Values should actually be different for the last case (the case with non-editable columns AND when `addRowEnabled` is true): `parameters` are patched to change the value of `addRowEnabled`, we should observe differences between the two objects.

Reply:
test fixed, done